RiotSharp/RiotApi.cs

```diff
@@ -1046,5 +1053,136 @@ public FeaturedGames GetFeaturedGames(Region region)
+        /// <returns>All champions mastery for summoner ID.</returns>
+        public async Task<List<ChampionMastery>> GetAllChampionsMasteryAsync(Platform platform, long summonerId)
+        {
+            var rootUrl = string.Format(ChampionMasteryRootUrl, platform, summonerId);
+
+            var json = await requester.CreateGetRequestAsync(rootUrl + ChampionMasteryAllChampions,
+                platform.ConvertToRegion());
+            return (await Task.Factory.StartNew(() => JsonConvert.DeserializeObject<List<ChampionMastery>>(json)));
+        }
+
+        /// <summary>
+        /// Get a player's total champion mastery score,
+        /// which is sum of individual champion mastery levels, by summoner ID synchronously.
+        /// </summary>
+        /// <param name="platform">Region where to retrieve the data.</param>
+        /// <param name="summonerId">ID of the summoner for which to retrieve champion mastery.</param>
+        /// <returns>Total champion mastery score for summoner ID.</returns>
+        public int GetTotalChampionMasteryScore(Platform platform, long summonerId)
+        {
+            var rootUrl = string.Format(ChampionMasteryRootUrl, platform, summonerId);
+
+            var json = requester.CreateGetRequest(rootUrl + ChampionMasteryTotalScore,
+                platform.ConvertToRegion());
+            return JsonConvert.DeserializeObject<int>(json);
+        }
+
+        /// <summary>
+        /// Get a player's total champion mastery score,
+        /// which is sum of individual champion mastery levels, by summoner ID asynchronously.
+        /// </summary>
+        /// <param name="platform">Region where to retrieve the data.</param>
+        /// <param name="summonerId">ID of the summoner for which to retrieve champion mastery.</param>
+        /// <returns>Total champion mastery score for summoner ID.</returns>
+        public async Task<int> GetTotalChampionMasteryScoreAsync(Platform platform, long summonerId)
+        {
+            var rootUrl = string.Format(ChampionMasteryRootUrl, platform, summonerId);
+
+            var json = await requester.CreateGetRequestAsync(rootUrl + ChampionMasteryTotalScore,
+                platform.ConvertToRegion());
+            return (await Task.Factory.StartNew(() => JsonConvert.DeserializeObject<int>(json)));
+        }
+
+        /// <summary>
+        /// Gets specified number of top champion mastery entries,
+        /// sorted by number of champion points descending, by summoner ID synchronously.
+        /// </summary>
+        /// <param name="platform">Region where to retrieve the data.</param>
+        /// <param name="summonerId">ID of the summoner for which to retrieve champion mastery.</param>
+        /// <param name="count">Number of entries to retrieve, defaults to 3.</param>
+        /// <returns>Top champions for summoner ID.</returns>
+        public List<ChampionMastery> GetTopChampionsMastery(Platform platform, long summonerId,
```

Comment:
same comment as above, I think this method should be renamed in order to expect a list.